src/Ombi.Core/Authentication/LdapUserManager.cs

```diff
@@ -0,0 +1,168 @@
+﻿using System;
+using System.Threading.Tasks;
+using Microsoft.Extensions.Logging;
+using Novell.Directory.Ldap;
+using Ombi.Core.Settings;
+using Ombi.Settings.Settings.Models;
+using Ombi.Store.Entities;
+
+namespace Ombi.Core.Authentication
+{
+    /// <summary>
+    /// Ldap Authentication Provider
+    /// </summary>
+    public class LdapUserManager : ILdapUserManager
+    {
+        public LdapUserManager(ILogger<LdapUserManager> logger, ISettingsService<LdapSettings> ldapSettings)
+        {
+            _ldapSettingsService = ldapSettings;
+            _logger = logger;
+        }
+
+        private readonly ISettingsService<LdapSettings> _ldapSettingsService;
+        private readonly ILogger<LdapUserManager> _logger;
+
+        public async Task<LdapSettings> GetSettings()
+        {
+            return await _ldapSettingsService.GetSettingsAsync();
+        }
+
+        public async Task<OmbiUser> LdapEntryToOmbiUser(LdapEntry entry)
+        {
+            var settings = await GetSettings();
+            var userName = GetLdapAttribute(entry, settings.UsernameAttribute).StringValue;
```

Comment:
This is going to throw a `NullReferenceException` by calling `.StringValue` on a null object